File: lab5datetime.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.time.temporal.ChronoUnit;

public class lab5datetime {

    // Function to calculate days between two LocalDate
    public static long getDaysBetween(LocalDate start, LocalDate end) {
        return ChronoUnit.DAYS.between(start, end);
    }

    public static void main(String[] args) {
        // Current Date and DateTime
        LocalDate currentDate = LocalDate.now();
        LocalDateTime currentDateTime = LocalDateTime.now();
        System.out.println("Current Date: " + currentDate);
        System.out.println("Current DateTime: " + currentDateTime);

        // Date parsing and formatting
        String dateStr = "2024-05-10";
        LocalDate parsedDate = LocalDate.parse(dateStr);
        System.out.println("Parsed Date: " + parsedDate);

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
        String formatted = parsedDate.format(formatter);
        System.out.println("Formatted Date: " + formatted);

        // Add/Subtract days
        LocalDate plusDays = parsedDate.plusDays(5);
        LocalDate minusDays = parsedDate.minusDays(3);
        System.out.println("Add 5 days: " + plusDays);
        System.out.println("Subtract 3 days: " + minusDays);

        // Get parts of date
        int year = parsedDate.getYear();
        int month = parsedDate.getMonthValue();
        int day = parsedDate.getDayOfMonth();
        System.out.println("Year: " + year + ", Month: " + month + ", Day: " + day);

        // Difference in days
        LocalDate endDate = LocalDate.of(2024, 5, 20);
        long diffDays = getDaysBetween(parsedDate, endDate);
        System.out.println("Difference between " + parsedDate + " and " + endDate + " is: " + diffDays + " days");

        // Difference using LocalDateTime
        LocalDateTime dt1 = LocalDateTime.of(2024, 5, 1, 10, 30);
        LocalDateTime dt2 = LocalDateTime.of(2024, 5, 14, 5, 0);
        long diffDateTime = ChronoUnit.DAYS.between(dt1, dt2);
        System.out.println("LocalDateTime difference: " + diffDateTime + " days");
    }
}

File: lab5datetime.py
from datetime import date, datetime, timedelta

# Function to calculate days between two dates
def get_days_between(start: date, end: date) -> int:
    return (end - start).days

# Current date and datetime
current_date = date.today()
current_datetime = datetime.now()
print("Current date:", current_date)
print("Current datetime:", current_datetime)

# Parse date string and format
date_str = "2024-05-10"
parsed_date = datetime.strptime(date_str, "%Y-%m-%d").date()
print("Parsed date:", parsed_date)

formatted = parsed_date.strftime("%d %b %Y")
print("Formatted date:", formatted)

# Add/Subtract days
plus_days = parsed_date + timedelta(days=5)
minus_days = parsed_date - timedelta(days=3)
print("Add 5 days:", plus_days)
print("Subtract 3 days:", minus_days)

# Get year/month/day
print(f"Year: {parsed_date.year}, Month: {parsed_date.month}, Day: {parsed_date.day}")

# Difference in days
end_date = date(2024, 5, 20)
diff_days = get_days_between(parsed_date, end_date)
print(f"Difference between {parsed_date} and {end_date} is: {diff_days} days")

# LocalDateTime difference
dt1 = datetime(2024, 5, 1, 10, 30)
dt2 = datetime(2024, 5, 14, 5, 0)
diff_datetime = (dt2 - dt1).days
print(f"datetime (LocalDateTime) difference: {diff_datetime} days")
